conans/client/conanfile/configure.py

```diff
@@ -33,9 +33,11 @@ def run_configure_method(conanfile, down_options, profile_options, ref):
 
     PackageType.compute_package_type(conanfile)
 
-    conanfile.build_requires = BuildRequirements(conanfile.requires)
-    conanfile.test_requires = TestRequirements(conanfile.requires)
-    conanfile.tool_requires = ToolRequirements(conanfile.requires)
+    conanfile._conan_add_requirement_allowed = True
```

Comment:
Doing this instead of `remove_conanfile_attr` context manager because I want to guard in every method